library/std/src/io/stdio.rs

```diff
@@ -873,6 +875,44 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Reads a [`String`] from [standard input](Stdin). The trailing
+/// newline is stripped. Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_input)]
+/// use std::io;
+///
+/// fn main() -> io::Result<()> {
+///     print!("Enter name: ");
+///
+///     let name: String = io::read_line()?;
+///
+///     println!("Your name is {}!", name);
+///
+///     Ok(())
+/// }
+/// ```
+#[unstable(
+    feature = "io_input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn read_line() -> Result<String> {
+    stdout().flush()?; // print!() does not flush :(
+    let mut input = String::new();
+    match stdin().read_line(&mut input)? {
+        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly")),
+        _ => Ok(input.trim_end_matches(&['\n', '\r'][..]).to_owned()),
```

Comment:
This now allocates a new string, copies all the data (except the newline) over to the new one, and throws away the old one (`input`).